### PR TITLE
ptunnel: init at 0.72

### PR DESCRIPTION
**Pull request by D3vil0p3r**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by a-n-n-a-l-e-e** [comment]
there is a https://github.com/utoni/ptunnel-ng which has more recent updates.

if staying with this tool, perhaps some of the patches used by the other distributions are relevant given this package hasn't been updated in over 10 years.

https://salsa.debian.org/alteholz/ptunnel/-/tree/master/debian/patches?ref_type=heads

the memset patch should be required, the others not so much.

**Comment by D3vil0p3r** [comment]
> there is a https://github.com/utoni/ptunnel-ng which has more recent updates.
> 
> if staying with this tool, perhaps some of the patches used by the other distributions are relevant given this package hasn't been updated in over 10 years.
> 
> https://salsa.debian.org/alteholz/ptunnel/-/tree/master/debian/patches?ref_type=heads
> 
> the memset patch should be required, the others not so much.

I already tried to apply those patches but I get errors...

**Comment by D3vil0p3r** [comment]
Furthermore, @a-n-n-a-l-e-e why on darwin I get the error:
```
> gcc -Wall -g `[ -e /usr/include/selinux/selinux.h ] && echo -DHAVE_SELINUX` -c -o ptunnel.o ptunnel.c
> /nix/store/vim1dldm8kflmy639w6ipmafm2sr5cby-bash-5.2p26/bin/bash: line 1: gcc: command not found
```
? gcc should not be already there due to `stdenv.mkDerivation`?

File: pkgs/by-name/pt/ptunnel/package.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  pname = "ptunnel";
+  version = "0.72";
+
+  src = fetchurl {
+    url = "https://www.cs.uit.no/~daniels/PingTunnel/PingTunnel-${finalAttrs.version}.tar.gz";
+    hash = "sha256-sxj3qn2IkYtiadBUp+JvBPl9iHD0e9Sadsssmcc0B6Q=";
+  };
+
+  makeFlags = [
+    "prefix=${placeholder "out"}"
```
Comment (a-n-n-a-l-e-e) [inline pkgs/by-name/pt/ptunnel/package.nix:17]:
```suggestion
    "prefix=$(out)"
    "CC=cc"
```

minor nit: placeholder "out" is fine, but in makeFlags, can use make syntax and just set `prefix=$(out)`.

there is no `gcc` on default darwin stdenv but can just set `CC=cc` and that will find the c compiler for darwin and linux

File: pkgs/by-name/pt/ptunnel/package.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ptunnel";
+  version = "0.72";
+
+  src = fetchurl {
+    url = "https://www.cs.uit.no/~daniels/PingTunnel/PingTunnel-${finalAttrs.version}.tar.gz";
+    hash = "sha256-sxj3qn2IkYtiadBUp+JvBPl9iHD0e9Sadsssmcc0B6Q=";
+  };
+
```
Comment (a-n-n-a-l-e-e) [inline pkgs/by-name/pt/ptunnel/package.nix:15]:
```suggestion
  patches = [
    (fetchpatch {
      url = "https://salsa.debian.org/alteholz/ptunnel/-/raw/1dbf9b69507e19c86ac539fd8e3c60fc274717b3/debian/patches/memset-fix.patch";
      hash = "sha256-dYbuMM0/ZUgi3OxukBIp5rKhlwAjGu7cl/3w3sWr/xU=";
    })
  ];
```

**Review by ghost — COMMENTED** [review]
need to add `fetchpatch` and perhaps some new lines between src, patches, makeFlags, and buildInputs :-)

File: pkgs/by-name/pt/ptunnel/package.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{ lib
+, stdenv
+, fetchurl
+, fetchpatch
```
Comment (a-n-n-a-l-e-e) [inline pkgs/by-name/pt/ptunnel/package.nix:4]:
```suggestion
, fetchpatch
```

**Review by ghost — APPROVED** [review]

**Comment by D3vil0p3r** [comment]
@a-n-n-a-l-e-e should we add also the other patch files from Salsa Debian?

**Comment by a-n-n-a-l-e-e** [comment]
> @a-n-n-a-l-e-e should we add also the other patch files from Salsa Debian?

i'd only add the memset patch. the others don't really matter.

**Comment by D3vil0p3r** [comment]
> > @a-n-n-a-l-e-e should we add also the other patch files from Salsa Debian?
> 
> i'd only add the memset patch. the others don't really matter.

Ah... I added also the other ones... Is it a problem?

**Comment by a-n-n-a-l-e-e** [comment]
> > > @a-n-n-a-l-e-e should we add also the other patch files from Salsa Debian?
> > 
> > 
> > i'd only add the memset patch. the others don't really matter.
> 
> Ah... I added also the other ones... Is it a problem?

you should remove the add_hardening_flags.diff and makefile-debian-prefix.patch because they are not doing anything. the others are at least fixing typos.

**Comment by D3vil0p3r** [comment]
> > > > @a-n-n-a-l-e-e should we add also the other patch files from Salsa Debian?
> > > 
> > > 
> > > i'd only add the memset patch. the others don't really matter.
> > 
> > 
> > Ah... I added also the other ones... Is it a problem?
> 
> you should remove the add_hardening_flags.diff and makefile-debian-prefix.patch because they are not doing anything. the others are at least fixing typos.

Fixed.

**Review by ghost — APPROVED** [review]

**Pull request merged**